### PR TITLE
trying to make permissions method similar to documents and deleted ids

### PR DESCRIPTION
**Pull request by mchernyavskaya**
## No issue exists for this

Small change to make the base call return something, so that it can be used with a `to_a` in specs, same as we do with the other base methods for documents and deleted ids.

I don't expect it to break anything.

### CODE REVIEW

**Review by artem-shelkovnikov — APPROVED** [review]

**Comment by artem-shelkovnikov** [comment]
It would be great to add a test to surface the problem that led to introducing the change, but change looks good to me!

**Comment by mchernyavskaya** [comment]
> It would be great to add a test to surface the problem that led to introducing the change, but change looks good to me!

I was trying to write a test for a new connector using subject.permissions(user_id).to_a =) to eager-load the yields. And it wasn't working because of no return value.

**Review by seanstory — APPROVED** [review]

File: lib/connectors_sdk/base/extractor.rb [hunk 1]
```diff
@@ -115,7 +115,7 @@ def with_auth_tokens_and_retry(&block)
         end
       end
 
-      def yield_document_changes(modified_since: nil)
+      def yield_document_changes(modified_since: nil, break_after_page: false)
```
Comment (wangch079) [inline lib/connectors_sdk/base/extractor.rb:118]: Why add `break_after_page` here?

With async flow in place, I think @artem-shelkovnikov has reverted all the `break_after_page` changes here: https://github.com/elastic/connectors/pull/101

**Review by wangch079 — COMMENTED** [review]
You may want to communicate to @RossAllenBell on whether you want to bring this back to `ent-search`
https://elastic.slack.com/archives/C01795T48LQ/p1652816751978499

File: lib/connectors_sdk/base/extractor.rb (hunk 1 above)
Comment (mchernyavskaya) [reply lib/connectors_sdk/base/extractor.rb]: It was a merge oversight. Removed; thanks.

**Comment by mchernyavskaya** [comment]
cc @RossAllenBell - bringing you here because of the comment from @wangch079 above.

**Review by wangch079 — APPROVED** [review]
>cc @RossAllenBell - bringing you here because of the comment from @wangch079 above.

@mchernyavskaya , @RossAllenBell had removed the dependency on `connectors_sdk` gem. If you want to bring this change to `ent-search`, you can do so directly by opening another PR in `ent-search` repo.

**Comment by RossAllenBell** [comment]
These changes could go either way, but I certainly don't think they would hurt. Why don't we get into the practice of copying them over to `ent-search` by default, including these changes.

**Pull request merged**